### PR TITLE
fix(switch): improve a11y and updates styles

### LINKED ISSUE #914: Switch enhancement: visual differentiation for on/off states
Currently, the visual styling of the on/off states of the switch could make it difficult for some users to differentiate the two states.

image

Some ideas of how to address this issue:

1. include some additional indicator that means “on” (similar to [Carbon](http://www.carbondesignsystem.com/components/toggle/code), [Salesforce](https://www.lightningdesignsystem.com/components/checkbox-toggle/), [Ant](https://ant.design/components/switch/))
2. use more of an opposite background color for "off"
3. display a tooltip on hover

### PR DESCRIPTION
**Pull request by christiemolloy**
Closes #914 and #915.

Fixes a11y, uses CSS to toggle the display of the label between "On" and "Off" and the color.

Adds icon to switch when it used without a label - refer to this conversation #1204, changes colors of switches to match what Michael Celedonia added to the issue.

### CODE REVIEW

File: src/patternfly/components/Switch/examples/switch-disabled-example.hbs [hunk 1]
```diff
@@ -1,16 +1,18 @@
-{{#> switch switch--attribute='for="switch-on-disabled-example"'}}
-  {{#> switch-input switch-input--attribute='type="checkbox" id="switch-on-disabled-example" aria-label="..." disabled checked'}}{{/switch-input}}
+{{#> switch switch--attribute='for="on-disabled"'}}
+  {{#> switch-input id="on-disabled" switch-input--attribute='type="checkbox" id="on-disabled" disabled checked'}}{{/switch-input}}
```
Comment (mcoker) [inline src/patternfly/components/Switch/examples/switch-disabled-example.hbs]: you're using this ID twice

File: src/patternfly/components/Switch/switch.scss [hunk 2]
```diff
@@ -100,7 +123,17 @@
     }
   }
 
-  .pf-c-switch__label {
+  .pf-c-switch__toggle-icon {
+    position: absolute;
+    top: var(--pf-c-switch__toggle-icon--Top);
+    display: block;
```
Comment (matthewcarleton) [inline src/patternfly/components/Switch/switch.scss]: do you need this?

File: src/patternfly/components/Switch/switch-label-off.hbs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<span class="pf-c-switch__label-off{{#if switch-label-off--modifier}} {{switch-label-off--modifier}}{{/if}}"
```
Comment (matthewcarleton) [inline src/patternfly/components/Switch/switch-label-off.hbs]: could we just have a single switch label that has a modifier for on/off?

File: src/patternfly/components/Switch/switch.scss [hunk 4]
```diff
@@ -100,7 +123,17 @@
...
+    padding-left: var(--pf-c-switch__toggle-icon--PaddingLeft);
+    font-size: var(--pf-c-switch__toggle-icon--FontSize);
+    color: var(--pf-c-switch__toggle-icon--Color);
+  }
+
+  .pf-c-switch__label-on,
```
Comment (matthewcarleton) [inline src/patternfly/components/Switch/switch.scss]: if we had a single switch with just a modifier we could simplify the way we write the css too.

File: src/patternfly/components/Switch/docs/code.md [hunk 5]
```diff
@@ -23,4 +23,5 @@ Use checkbox if your user has to perform additional steps for changes to become
 | `.pf-c-switch` | `<label>` |  Initiates a switch. **required**  |
 | `.pf-c-switch__input` | `<input type="checkbox">` |  Hide the checkbox inside the switch. **required**  |
 | `.pf-c-switch__toggle` | `<span>` |  Initiates the toggle inside the switch. **required**  |
+| `.pf-c-switch__toggle-icon` | `<i>` | Initiates an icon inside the switch toggle. **required when the switch is used without a label** |
 | `.pf-c-switch__label` | `<span>` |  Initiates a label inside the switch. |
```
Comment (matthewcarleton) [inline src/patternfly/components/Switch/docs/code.md]: We'll have to update the docs to include the new label approach.

File: src/patternfly/components/Switch/switch.scss [hunk 6]
```diff
@@ -20,6 +25,14 @@
   --pf-c-switch__toggle--before--BackgroundColor: var(--pf-global--BackgroundColor--light-100);
   --pf-c-switch__toggle--before--BorderRadius: var(--pf-c-switch__toggle--before--Width);
```
Comment (matthewcarleton) [inline src/patternfly/components/Switch/switch.scss]: can you change the value of this to `--pf-global--BorderRadius--lg` it will fix that bug in chrome.

**Pull request merged**